src/components/Tamagotchi.tsx

```diff
@@ -63,6 +67,33 @@ const Tamagotchi: React.FC = () => {
       clearInterval(pollInterval);
     }
   };
+
+  const handleBath = async () => {
+    setIsInteracting(true);
+    setTamaStatus("Bathing...");
+    try {
+      const response = await fetch("/api/interact", {
+        method: "POST",
+        body: JSON.stringify({
+          interactionType: INTERACTION.BATH,
+        }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      const responseText = await response.text();
+      handleResponse(responseText);
+    } catch (e) {
+      console.log(e);
+    }
+
+    setTimeout(() => {
+      setAnimation(idle);
+      setTamaStatus(DEFAULT_STATUS);
+      setIsInteracting(false);
+    }, 9000);
```

Comment:
**suggestion:** Consider making the animation duration configurable

Extract magic number 9000 to a named constant or configuration value

```
    const ANIMATION_DURATION_MS = 9000;
    }, ANIMATION_DURATION_MS);
```